wm323Sup/wm323.db

```diff
@@ -0,0 +1,185 @@
+record(bo, "$(P)SIM")
+{
+    field(SCAN, "Passive")
+    field(DTYP, "Soft Channel")
+    field(ZNAM, "NO")
+    field(ONAM, "YES")
+    field(VAL, "$(RECSIM=0)")
+    field(PINI, "YES")
+}
+
+record(bo, "$(P)DISABLE")
+{
+    field(DESC, "Disable comms")
+    field(PINI, "YES")
+    field(VAL, "$(DISABLE=0)")
+    field(OMSL, "supervisory")
+    field(ZNAM, "COMMS ENABLED")
+    field(ONAM, "COMMS DISABLED")
+}
+
+record(longin, "$(P)SPEED")
+{
+    field(DESC, "Current pump speed")
+    field(SCAN, "1 second")
+    field(DTYP, "stream")
+  
+    field(INP, "@wm323.proto getSpeed $(PORT)")
+  
+    field(EGU, "rpm")
+  
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:SPEED")
+    field(SDIS, "$(P)DISABLE")
+    info(INTEREST, "HIGH")
+    info(archive, "VAL")
+}
+
+record(longout, "$(P)SPEED:SP")
+{
+    field(DESC, "Pump speed set point")
+    field(SCAN, "Passive")
+    field(DTYP, "stream")
+  
+    field(OUT, "@wm323.proto setSpeed $(PORT)")
+  
+    field(EGU, "rpm")
+    field(DRVL, 3)
+    field(DRVH, 400)
+
+    info(INTEREST, "HIGH")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:SPEED:SP")
+    field(SDIS, "$(P)DISABLE")
+}
+
+record(bi, "$(P)DIRECTION")
+{
+    field(DESC, "Pump rotation direction")
+    field(SCAN, "1 second")
+    field(DTYP, "stream")
+
+    field(INP, "@wm323.proto getDirection $(PORT)")
+
+    field(ZNAM, "Anti-Clockwise")
+    field(ONAM, "Clockwise")
+
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:DIRECTION")
+    field(SDIS, "$(P)DISABLE")
+    info(INTEREST, "HIGH")
+    info(archive, "VAL")
+}
+
+record(bo, "$(P)DIRECTION:SP")
+{
+    field(DESC, "Pump direction set point")
+    field(SCAN, "Passive")
+    field(DTYP, "stream")
+
+    field(OUT, "@wm323.proto setDirection $(PORT)")
+
+    field(ZNAM, "Anti-Clockwise")
+    field(ONAM, "Clockwise")
+
+    info(INTEREST, "HIGH")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:DIRECTION:SP")
+    field(SDIS, "$(P)DISABLE")
+}
+
+record(bi, "$(P)STATUS")
+{
+    field(DESC, "Pump running status")
+    field(SCAN, "1 second")
+    field(DTYP, "stream")
+
+    field(INP, "@wm323.proto getStatus $(PORT)")
+
+    field(ZNAM, "Stopped")
+    field(ONAM, "Running")
+
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:STATUS")
+    field(SDIS, "$(P)DISABLE")
+    info(INTEREST, "HIGH")
+    info(archive, "VAL")
+}
+
+record(bo, "$(P)STATUS:SP") {
+    field(DESC, "Pump running status setpoint")
+    field(SCAN, "Passive")
+    field(DTYP, "stream")
+    
+    field(OUT, "@wm323.proto setStatus $(PORT)")
+    
+    field(ZNAM, "Stopped")
+    field(ONAM, "Running")
+    
+    info(INTEREST, "HIGH")
+    field(SIML, "$(P)SIM")
+    field(SIOL, "$(P)SIM:STATUS:SP")
+    field(SDIS, "$(P)DISABLE")
+}
+
+record(bo, "$(P)RUN:SP")
```

Comment:
Just a side note: next time you do something like this, to avoid duplicating the protocol, you can actually get the `RUN:SP` and `STOP:SP` records to write 0 or 1 to the usual `STATUS:SP` record. This avoids a little bit of duplication. I don't think it's a major worry here though, as this device is relatively simple, so I'm happy for you to leave it as-is for now.